src/main/drivers/compass/compass_ak8963.c

```diff
@@ -368,6 +368,7 @@ static bool ak8963Init(magDev_t *mag)
 
     // Trigger first measurement
     ak8963WriteRegister(dev, AK8963_MAG_REG_CNTL1, CNTL1_BIT_16_BIT | CNTL1_MODE_ONCE);
+    mag->magOdrHz = 50; // arbitrary value, need to check what ODR is actually returned
```

Comment:
Single measurement mode is used. Data is available 9ms (worst case) after wr9iting CNTL_MODE_ONCE. IMO this value should be `0` (compassReadIntervalUs = TASK_PERIOD_HZ(TASK_COMPASS_RATE_HZ); below)

Reply:
To be honest I find the AK8963 driver very confusing.  Is it in MPU_SLAVE mode?  Of single read mode?  The behaviour seem different.
If it is in `MPU_SLAVE` mode, that seems to have multiple states, and they should be handled properly if we use the current code.  
If it is in single read mode, then it looks like when we do any read, it would return the data.  Does that mean that the check in `checkReadState` would cause the data to be lost?  Also I noticed a `return false` in `ak8963DirectReadData()` which perhaps could need to be handled.
How should I best manage this?  Do we need different behaviour if we use the Direct Read / single read method?

Reply:
IMO both modes are single-shot. But ak8963SlaveReadData handles I2C timing (need to wait until read completes). 
So you need fee calls:
1. Write start of conversion
2. read status register (After ODR delay)
3. check content1 of status register (status1 I2C read time) / stat read of data
4. process data (7byte I2C read time)

See below how to reduce number of reads.
Also - slave mode call delayMicroseconds(10); multiple times. It may be better to simply ignore this MAG until someone fixes it.

Reply:
I kind of get the feeling we should deprecate support for this mag and the other AK8975 as well.  They are both very old and can't be purchased anymore.  
Or, alternatively, we can wait and see if people have issues with them.  If no-one has issues, then either they are not being used 'in the wild', or the driver sort of works adequately.
I don't have the hardware to test; perhaps if someone has it, and has a problem, and is willing to help, we could maintain support for these old units.
But overall we should deprecate them sometime soon.
I will make a note in the documentation that there may be issues with these mags and that they are unsupported.